client/my-sites/checkout/composite-checkout/wpcom/hooks/use-shopping-cart.ts

```diff
@@ -249,6 +252,24 @@ function shoppingCartHookReducer(
 	return state;
 }
 
+function removeItemFromLocalStorage( productSlug: string ) {
+	const cartItemsFromLocalStorage = JSON.parse(
+		window.localStorage.getItem( 'shoppingCart' ) || '[]'
+	);
+
+	if ( ! isEmpty( cartItemsFromLocalStorage ) ) {
```

Comment:
I don't think we need to use the heavy lifting power of `isEmpty` here. `JSON.parse()` can return a lot of different types, but we only want to continue here if `cartItemsFromLocalStorage` is an array, so perhaps this could be `if ( Array.isArray( cartItemsFromLocalStorage ) )` instead.

That wouldn't prevent the block from running if it's an empty array, but `newCart` would be fine in that case, it would just be another empty array and it's probably fine to save that empty array again. If you'd prefer not to save the empty array, we could just add a `if ( newCart.length > 0 )` guard before the localstorage setter.

To avoid having four levels of indentation, the first two guards could be changed to early returns instead.

Reply:
All good points, updated with these changes.